api/src/paths/project/{projectId}/survey/{surveyId}/observation/submission/get.ts

```diff
@@ -165,26 +165,38 @@ export function getOccurrenceSubmission(): RequestHandler {
         return res.status(200).json(null);
       }
 
-      const hasAdditionalOccurrenceSubmissionMessages =
+      const willFetchAdditionalMessages =
         occurrenceSubmission.submission_status_type_name &&
         [
-          SUBMISSION_STATUS_TYPE.REJECTED,
+          // Submission statuses for validation/transformation failure
           SUBMISSION_STATUS_TYPE.SYSTEM_ERROR,
           SUBMISSION_STATUS_TYPE.FAILED_OCCURRENCE_PREPARATION,
           SUBMISSION_STATUS_TYPE.FAILED_VALIDATION,
           SUBMISSION_STATUS_TYPE.FAILED_TRANSFORMED,
-          SUBMISSION_STATUS_TYPE.FAILED_PROCESSING_OCCURRENCE_DATA
+          SUBMISSION_STATUS_TYPE.FAILED_PROCESSING_OCCURRENCE_DATA,
+          SUBMISSION_STATUS_TYPE['AWAITING CURRATION'],
+          SUBMISSION_STATUS_TYPE.REJECTED,
+          SUBMISSION_STATUS_TYPE['ON HOLD']
         ].includes(occurrenceSubmission.submission_status_type_name);
 
-      const messageTypes: IMessageTypeGroup[] = hasAdditionalOccurrenceSubmissionMessages
+      const isDoneValidating =
+        willFetchAdditionalMessages ||
+        (occurrenceSubmission.submission_status_type_name &&
+          [
+            // Submission statuses for validation completion
+            SUBMISSION_STATUS_TYPE.TEMPLATE_VALIDATED,
+            SUBMISSION_STATUS_TYPE.DARWIN_CORE_VALIDATED
```

Comment:
We may want/need to introduce a new status, that represents a generic completed state. Then we can just check if the submission is in that state to know if it is fully complete or not.

Reply:
Technically a template isn't fully complete until it has been xlsx validated, transformed, and dwc validated.

But, in future if we accept DWC submissions from the user, we obv won't run the first 2 steps, and it will be compete when just the dwc validation has run.

Reply:
> We may want/need to introduce a new status, that represents a generic completed state. Then we can just check if the submission is in that state to know if it is fully complete or not.

Agreed 🤔